### PR TITLE
[Merged by Bors] - feat: port Data.Nat.Digits

### PR DESCRIPTION
**Pull request by qawbecrdtey**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by dwrensha** [comment]
Ready for review. I commented out the tactic stuff at the bottom for now. (We've commented out similar meta code in [Fib.lean](https://github.com/leanprover-community/mathlib4/blob/7ca25b3affcf79c45bc1f3159aa840b4ecb6ad02/Mathlib/Data/Nat/Fib.lean#L396-L399) and [Int/GCD.lean](https://github.com/leanprover-community/mathlib4/blob/7ca25b3affcf79c45bc1f3159aa840b4ecb6ad02/Mathlib/Data/Int/GCD.lean#L512-L518).)

File: Mathlib/Data/Nat/Digits.lean [hunk 1]
```diff
@@ -549,92 +538,96 @@ theorem ofDigits_zmod (b : ℤ) (k : ℕ) (L : List ℕ) : ofDigits b L % k = of
...
-  have t := dvd_iff_dvd_of_digits 11 10 (-1 : ℤ) (by norm_num) n
-  rw [of_digits_neg_one] at t
+  have t := dvd_iff_dvd_ofDigits 11 10 (-1 : ℤ) (by norm_num) n
+  rw [ofDigits_neg_one] at t
   exact t
 #align nat.eleven_dvd_iff Nat.eleven_dvd_iff
 
+-- Porting note: Is this proved somewhere in mathlib?
+theorem pow_zero_ {n : ℤ} : n ^ 0 = 1 := by simp
+theorem pow_succ_ {n : ℤ} {p : ℕ} : n ^ (succ p) = n * (n ^ p) := by
```
Comment (jcommelin) [inline Mathlib/Data/Nat/Digits.lean:619]: They should simply be `pow_zero` and `pow_succ`. Would you need an extra import for those?
It looks like `pow_succ` was available here in mathlib3.

**Review by jcommelin — COMMENTED** [review]
Modulo `pow_succ_` lgtm

bors d+

File: Mathlib/Data/Nat/Digits.lean (hunk 1 above)
Comment (dwrensha) [reply Mathlib/Data/Nat/Digits.lean:619]: Hm. Looks like the problem is that [`pow_succ`](https://github.com/leanprover-community/mathlib4/blob/a0a7b109e62bfa3d0f3355091cd112928f398d67/Mathlib/Algebra/Group/Defs.lean#L630) is getting shadowed by the more specialized [`Nat.pow_succ`](https://github.com/leanprover/lean4/blob/3f6c5f17db30c2a2dcaca1b9e9faec4e2cf04b95/src/Init/Data/Nat/Basic.lean#L473). It works if I fully qualify the `pow_succ` here as `_root_.pow_succ`.

**Pull request closed without merging**